code/game/objects/items/weapons/RMS_vr.dm

```diff
@@ -0,0 +1,231 @@
+	spark_system.set_up(5, 0, src)
+	spark_system.attach(src)
+	update_icon()
+
+/obj/item/weapon/pipe_dispenser/Destroy()
+	qdel_null(spark_system)
+	return ..()
+
+/obj/item/weapon/rms/update_icon()
+	charge_stage = round((stored_charge/max_charge)*4)
+	if(charge_stage >= 4)
+		charge_stage = 4
+	add_overlay("rms_charge[charge_stage]")
+
+/obj/item/weapon/rms/examine(mob/user)
+	. = ..()
+	. += display_resources()
+
+/obj/item/weapon/rms/proc/display_resources()
+	return "It currently holds [round(stored_charge/1000)]/[max_charge/1000] kW charge."
+
+/obj/item/weapon/rms/proc/drain_battery(user, battery)
+	var/obj/item/weapon/cell.C = battery
+	if(stored_charge == max_charge)
+		to_chat(user, "<span class='notice'>The Rapid Material Synthesizer is full on charge!.</span>")
+		update_icon()
+		return
+	if(C.charge == 0)
+		to_chat(user, "<span class='notice'>The battery has no charge.</span>")
+		update_icon()
+		return
+	else
+		playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
+		if(do_after(user, 2,target = C))
+			stored_charge += C.charge
+			C.charge = 0
+			C.update_icon()
+			to_chat(user, "<span class='notice'>You drain [C].</span>")
+			if(stored_charge > max_charge)
+				stored_charge = max_charge
+			update_icon()
+			return
+
+/obj/item/weapon/rms/proc/consume_resources(amount)
+	stored_charge -= amount
+	update_icon()
+	return
+
+
+/obj/item/weapon/rms/proc/can_afford(amount)
+	if(mode_index != modes.len)
+		if(stored_charge >= charge_cost)
+			return TRUE
+		else
+			return FALSE
+	else
+		if(stored_charge >= charge_cost_r)
+			return TRUE
+		else
+			return FALSE
+
+/obj/item/weapon/rms/proc/use_rms(atom/A, mob/living/user)
+	var/obj/product
+	if(!can_afford())
+		to_chat(user, "<span class='notice'>There is not enough charge to use this mode.</span>")
+		return
+	playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+	if(do_after(user, 5, target = A))
+		switch(mode_index)
+			if(1)
+				if(charge_stage >= 4)
+					product = new /obj/item/stack/material/plasteel  // can only create one sheet at max charge, and uses five times more charge
+					consume_resources(charge_cost_m)
+					spark_system.start()
+				else
+					product = new /obj/item/stack/material/steel
+					consume_resources(charge_cost)
+			if(2)
+				if(charge_stage >= 4)
+					product= new /obj/item/stack/material/glass/phoronglass // can only create one sheet at max charge, and uses five times more charge
+					consume_resources(charge_cost_m)
+					spark_system.start()
+				else
+					product = new /obj/item/stack/material/glass
+					consume_resources(charge_cost)
+			if(3)
+				switch(charge_stage)
+					if(4)
+						product = pick(50; new /obj/item/weapon/cell/secborg/empty,
+										20; new /obj/item/weapon/cell/high/empty,
+										15; new /obj/item/weapon/cell/super/empty,
+										10; new /obj/item/weapon/cell/hyper/empty,
+										4; new /obj/item/device/fbp_backup_cell,
+										1; new /obj/item/weapon/cell/infinite)
```

Comment:
MMMM. I dunnnoooooooo. I mean you can brute force this every shift without tons of issue.

Reply:
True, but given that the cell actually isn't too powerful(it doesn't recharge by itself, it needs to be in a device to actually start its regeneration), and the abysmal chance of it appearing along with the device at full charge, its more of a gimmick than anything.  Plus, I believe that the time investment in even trying to get this outweighs the shenanigans.